contk/_utils/file_utils.py

```diff
@@ -0,0 +1,173 @@
+"""
+Utilities for working with the local dataset cache.
+This file is adapted from the AllenNLP library at https://github.com/allenai/allennlp
+Copyright by the AllenNLP authors.
+"""
+
+import os
+import logging
+import json
+import tempfile
+import shutil
+import hashlib
+
+from tqdm import tqdm
+from checksumdir import dirhash
+
+import requests
+
+class Downloader():
+	r'''Base class of Downloader
+	'''
+	def __init__(self):
```

Comment:
If there is only one Downloader (singleteon), write functions in module level. (Python's module is naturally singleton)